### PR TITLE
fixes #32494 - Removing fact parser

### PR DESCRIPTION
**Pull request by domitea**
Removing fact parser according to moving them to core

This a plugin part of https://github.com/theforeman/foreman/pull/8491 PR where I add a Ansible parser to core.

### CODE REVIEW

**Comment by tbrisker** [comment]
This is missing dropping of the AnsibleFactParser itself

File: lib/foreman_ansible/engine.rb [hunk 1]
```diff
@@ -76,13 +76,8 @@ class Engine < ::Rails::Engine
       if Rails.env.test? ||
          foreman_version.major.to_i == 1 && foreman_version.minor.to_i < 13
         ::Foreman::Plugin.fact_importer_registry.register(:ansible, ForemanAnsible::FactImporter)
```
Comment (tbrisker) [inline lib/foreman_ansible/engine.rb:78]: I'm pretty sure we can drop this as well, foreman_ansible requires a much newer foreman already

**Review by tbrisker — CHANGES_REQUESTED** [review]
We should also update the minimal foreman version and drop the legacy non-structured fact importer

File: lib/foreman_ansible/register.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 Foreman::Plugin.register :foreman_ansible do
-  requires_foreman '>= 2.2'
+  requires_foreman '>= 2.5'
```
Comment (tbrisker) [inline lib/foreman_ansible/register.rb:4]: I think this should be 2.6, since the parser was only recently merged in core

**Comment by ezr-ondrej** [comment]
[test foreman_ansible]

**Comment by tbrisker** [comment]
[test foreman_ansible]

**Review by ezr-ondrej — APPROVED** [review]
Jenkins is broken, but GH actions approves.
@adamruzicka anything to add here?

**Review by adamruzicka — CHANGES_REQUESTED** [review]
Could you also go through `.rubocop_todo.yml` and drop the no-longer-relevant lines from there?

**Comment by tbrisker** [comment]
[test foreman_ansible] now lets see if the fix in core worked...

**Comment by tbrisker** [comment]
Perhaps this pr needs a rebase?
```
LoadError: cannot load such file -- foreman_tasks_core
/usr/local/rvm/gems/ruby-2.7.0@foreman_ansible-pull-request-1/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'
/usr/local/rvm/gems/ruby-2.7.0@foreman_ansible-pull-request-1/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'
/usr/local/rvm/gems/ruby-2.7.0@foreman_ansible-pull-request-1/gems/activesupport-6.0.3.7/lib/active_support/dependencies.rb:324:in `block in require'
/usr/local/rvm/gems/ruby-2.7.0@foreman_ansible-pull-request-1/gems/activesupport-6.0.3.7/lib/active_support/dependencies.rb:291:in `load_dependency'
/usr/local/rvm/gems/ruby-2.7.0@foreman_ansible-pull-request-1/gems/activesupport-6.0.3.7/lib/active_support/dependencies.rb:324:in `require'
/home/jenkins/workspace/foreman_ansible-pull-request/database/postgresql/label/fast/ruby/2.7/plugin/lib/foreman_ansible_core.rb:3:in `<top (required)>'
/usr/local/rvm/gems/ruby-2.7.0@foreman_ansible-pull-request-1/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'
/usr/local/rvm/gems/ruby-2.7.0@foreman_ansible-pull-request-1/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'
/usr/local/rvm/gems/ruby-2.7.0@foreman_ansible-pull-request-1/gems/activesupport-6.0.3.7/lib/active_support/dependencies.rb:324:in `block in require'
/usr/local/rvm/gems/ruby-2.7.0@foreman_ansible-pull-request-1/gems/activesupport-6.0.3.7/lib/active_support/dependencies.rb:291:in `load_dependency'
/usr/local/rvm/gems/ruby-2.7.0@foreman_ansible-pull-request-1/gems/activesupport-6.0.3.7/lib/active_support/dependencies.rb:324:in `require'
/home/jenkins/workspace/foreman_ansible-pull-request/database/postgresql/label/fast/ruby/2.7/plugin/lib/foreman_ansible/engine.rb:7:in `<top (required)>'
/usr/local/rvm/gems/ruby-2.7.0@foreman_ansible-pull-request-1/gems/activesupport-6.0.3.7/lib/active_support/dependencies.rb:324:in `require'
/usr/local/rvm/gems/ruby-2.7.0@foreman_ansible-pull-request-1/gems/activesupport-6.0.3.7/lib/active_support/dependencies.rb:324:in `block in require'
/usr/local/rvm/gems/ruby-2.7.0@foreman_ansible-pull-request-1/gems/activesupport-6.0.3.7/lib/active_support/dependencies.rb:291:in `load_dependency'
/usr/local/rvm/gems/ruby-2.7.0@foreman_ansible-pull-request-1/gems/activesupport-6.0.3.7/lib/active_support/dependencies.rb:324:in `require'
/home/jenkins/workspace/foreman_ansible-pull-request/database/postgresql/label/fast/ruby/2.7/plugin/lib/foreman_ansible.rb:4:in `<top (required)>'

**Comment by ezr-ondrej** [comment]
[test foreman_ansible]

**Comment by ezr-ondrej** [comment]
Yeap, rebase would help, I'd merge without it, but Adam's point is valid, no reason to keep the lines in `.rubocop_todo.yml`

**Comment by tbrisker** [comment]
[test foreman_ansible] Should be green now, the issue in rex was fixed in https://github.com/theforeman/foreman_remote_execution/commit/e492aef4b0de384011990b7ddb28447dd3d3c8bf
GH actions tests should be fixed by https://github.com/theforeman/foreman_ansible/pull/430

**Review by tbrisker — APPROVED** [review]
Thanks @domitea, test failures are unrelated, lets get this in.

**Pull request merged**